### PR TITLE
[Feature] histogram statistics support collect top-n data

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: be/src/exprs/agg/histogram.h [hunk 1]
```diff
@@ -120,55 +89,157 @@ class HistogramAggregationFunction final
...
+            }
+        }
+        frequency_vec.emplace_back(std::make_pair(last_value, count));
+        //Sort by frequency
+        std::sort(frequency_vec.begin(), frequency_vec.end(), CmpPair<T>());
+
+        int top_n_size = std::min((int)ColumnHelper::get_const_value<TYPE_INT>(ctx->get_constant_column(3)),
+                                  (int)frequency_vec.size());
+        //Build a top-n map so that values are excluded from subsequent buckets
+        std::map<T, int> topn;
```
Comment (kangkaisen) [inline be/src/exprs/agg/histogram.h:131]: under_map

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 33 / 67 (49.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticAutoCollector.java|0|1|00.00%|[52]|
|:large_blue_circle:|com/starrocks/statistic/SampleStatisticsCollectJob.java|0|2|00.00%|[47, 50]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|0|12|00.00%|[39, 40, 68, 69, 112, 113, 115, 116, 117, 118, 119, 121]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|0|2|00.00%|[103, 122]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|0|1|00.00%|[344]|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|0|1|00.00%|[42]|
|:large_blue_circle:|com/starrocks/statistic/TableCollectJob.java|0|1|00.00%|[30]|
|:large_blue_circle:|com/starrocks/statistic/HistogramStatisticsCollectJob.java|0|7|00.00%|[36, 37, 38, 41, 64, 65, 66]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|4|8|50.00%|[59, 61, 62, 63]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|2|3|66.67%|[63]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|18|20|90.00%|[190, 201]|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|8|8|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java [hunk 2]
```diff
@@ -105,4 +107,17 @@ List<Bucket> convert(String histogramString) {
         }
         return buckets;
     }
+
+    private Map<Double, Long> convertTOPN(String histogramString) {
+        JsonObject jsonObject = JsonParser.parseString(histogramString).getAsJsonObject();
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java:112]: Yes, there is no need to deal with it here. 1. The data is generated internally, and there will be no parsing errors under normal circumstances. 2. There is an except catch in outer code

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 33 / 67 (49.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticAutoCollector.java|0|1|00.00%|[52]|
|:large_blue_circle:|com/starrocks/statistic/SampleStatisticsCollectJob.java|0|2|00.00%|[47, 50]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|0|12|00.00%|[39, 40, 68, 69, 112, 113, 115, 116, 117, 118, 119, 121]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|0|2|00.00%|[103, 122]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|0|1|00.00%|[344]|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|0|1|00.00%|[42]|
|:large_blue_circle:|com/starrocks/statistic/TableCollectJob.java|0|1|00.00%|[30]|
|:large_blue_circle:|com/starrocks/statistic/HistogramStatisticsCollectJob.java|0|7|00.00%|[36, 37, 38, 41, 64, 65, 66]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|4|8|50.00%|[59, 61, 62, 63]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|2|3|66.67%|[63]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|18|20|90.00%|[190, 201]|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|8|8|100.00%|[]|

**Review by Youngwb — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**